Dynamic Wallet Rupee Icon Color

shadeRed opened the pull request:
***Let me preface this with the fact that I am absolutely not at all fluent in c/c++. I just look at examples and imitate. I have no idea what's considered good or bad practices, but what I did here works and that's all that I'm really able to evaluate.***

Recolors the bottom left rupee icon according to what wallet you currently have:
![green](https://user-images.githubusercontent.com/30968528/160961229-482eedc7-6e10-4f4c-9564-9d4ee82fba42.PNG)
![blue](https://user-images.githubusercontent.com/30968528/160961237-4f02da2b-296d-4c35-ab14-e336a3ceb8d7.PNG)
![red](https://user-images.githubusercontent.com/30968528/160961241-33bbc388-95ef-4b33-9120-542093e7953c.PNG)

Green RGB: 200, 255, 100 (what was there by default)
Blue RGB: 100, 100, 255
Red RGB: 255, 100, 100

I know Ocarina of Time Randomizer implemented this as well as (I think) Majora's Mask, but I couldn't really figure out what RGB values that they were using for the colors. But I think for now, this is good enough. It can easily be changed later.

This can be turned on and off through the "Dynamic Wallet Icon" cvar option within `Enhancements` > `Graphics`

Colbydude commented:
> I know Ocarina of Time Randomizer implemented this as well as (I think) Majora's Mask, but I couldn't really figure out what RGB values that they were using for the colors. But I think for now, this is good enough. It can easily be changed later.

Values from the randomizer can be found at
https://github.com/TestRunnerSRL/OoT-Randomizer/blob/Dev/ASM/c/hud_colors.c
```c
const colorRGB8_t rupee_colors[4] = {
  { 0xC8, 0xFF, 0x64 }, // Base Wallet (Green)
  { 0x82, 0x82, 0xFF }, // Adult's Wallet (Blue)
  { 0xFF, 0x64, 0x64 }, // Giant's Wallet (Red)
  { 0xFF, 0x5A, 0xFF }, // Tycoon's Wallet (Purple)
};
```

Just a little off on the blue. :)
Thanks for adding!

shadeRed commented:
Awesome. I updated the blue rupee color and refactored the code a bit.

![image](https://user-images.githubusercontent.com/30968528/161137079-88eb0a44-d4b8-4175-835b-fb4fcfe15c47.png)

shadeRed commented:
resolved a conflict caused by the merge of #102

louist103 commented on soh/src/code/z_parameter.c:
```diff
@@ -3125,6 +3125,15 @@ void Interface_Draw(GlobalContext* globalCtx) {
     static s16 rupeeDigitsCount[] = { 2, 3, 3 };
+
+    // courtesy of https://github.com/TestRunnerSRL/OoT-Randomizer/blob/Dev/ASM/c/hud_colors.c
+    static s16 rupeeWalletColors[4][3] = {
+        { 0xC8, 0xFF, 0x64 }, // Base Wallet (Green)
+        { 0x82, 0x82, 0xFF }, // Adult's Wallet (Blue)
+        { 0xFF, 0x64, 0x64 }, // Giant's Wallet (Red)
+        { 0xFF, 0x5A, 0xFF }, // Tycoon's Wallet (Purple)
+    };
```
```suggestion
    static s16 rupeeWalletColors[4][3] = {
        { 0xC8, 0xFF, 0x64 }, // Base Wallet (Green)
        { 0x82, 0x82, 0xFF }, // Adult's Wallet (Blue)
        { 0xFF, 0x64, 0x64 }, // Giant's Wallet (Red)
    };
```
We dont have that wallet so we dont need this one.

InfoManiac742 replied:
Shouldn't the dimensions be [3][3] as well?

louist103 replied:
yes they should be updated.

louist103 commented on soh/src/code/z_parameter.c:
```diff
@@ -3167,7 +3176,15 @@ void Interface_Draw(GlobalContext* globalCtx) {
         if (fullUi) {
             // Rupee Icon
-            gDPSetPrimColor(OVERLAY_DISP++, 0, 0, 200, 255, 100, interfaceCtx->magicAlpha);
+            s16 *rColor;
```
```suggestion
            s16* rColor;
```
OOT styling.

louist103 reviewed (changes requested):
Nice work on this.  Just two small things.

shadeRed commented:
Gotcha. Requested changes have been made.

shadeRed commented:
(another quick commit. I forgot to update `rupeeWalletColors` dimensions. should be good to go now`)

Kenix3 reviewed (approved):
Merge pending @louist103's approval

NEstelami commented:
Merge pending rebase conflict resolution.

FitzRoyX commented:
This seems confusing to me. The rupee counter is green because rupees are denominated by color and only green is worth 1. Having 200 blue rupees would mean you have 1000 rupees.

shadeRed commented:
> This seems confusing to me. The rupee counter is green because rupees are denominated by color and only green is worth 1. Having 200 blue rupees would mean you have 1000 rupees.

Some feel as though it makes you feel the "progression" of obtaining bigger wallets. And I feel like it's pretty common practice in Zelda nowadays. Wind Waker for example implemented it where it went `green > blue > orange`. And it'll ultimately be an optional enhancement for those who wish to enable it.
